### PR TITLE
chore: add pr template

### LINKED ISSUE #3758: make PR template

### PR DESCRIPTION
**Pull request by MicaiahReid**
## PR description

This PR introduces a PR template that will be used to guide contributors on a format for creating PRs in the Ganache repo. Fixes #3758 

We can bike shed this to our heart's content, but I figured I'd get something out to start the conversation. By the way, this PR description uses the template. 

## Testing instructions

N/A

## Documentation

- [ ] I've added the `doc-change-required` label to this PR so that documentation can be added by the docs team in another PR.
- [ ] I've included documentation changes in this PR:
  - [ ] I've updated the `README.md`.
  - [ ] I've updated the `CONTRIBUTING.md`.
  - [ ] I've added/updated an RPC method in `src/chains/ethereum/ethereum/RPC-METHODS.md`.
  - [ ] I've added/updated a JS-DOC example of a new RPC method in `src/chains/ethereum/ethereum/src/api.ts`.
- [X] This PR does not require documentation changes.
- [x] I don't know what this PR needs in regards to documentation. 

## Release Notes

This change introduces a PR template that will be used to guide contributors on a format for creating PRs in the Ganache repo.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+<!-- Enter PR description here. Don't forget to link any issues that are resolved by this PR. -->
+
+## Testing instructions
+
+<!-- If there are any relevant instructions for testing this PR, add them here. -->
+
+## Documentation
+
+<!-- Consider the documentation needs of this PR, and select the appropriate option. -->
```
Comment (jeffsmale90) [inline .github/pull_request_template.md]: Maybe make this _not_ a comment, as it's not immediately clear that only one of these is required.

I always find these sort of "optional" checkboxes to be confusing...

**Review by jeffsmale90 — CHANGES_REQUESTED** [review]
I love it!

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+- [ ] I've added the `doc-change-required` label to this PR so that documentation can be added by the docs team in another PR.
+- [ ] I've included documentation changes in this PR:
+  - [ ] I've updated the `README.md`.
+  - [ ] I've updated the `CONTRIBUTING.md`.
+  - [ ] I've added/updated an RPC method in `src/chains/ethereum/ethereum/RPC-METHODS.md`.
```
Comment (jeffsmale90) [inline .github/pull_request_template.md:22]: TIL - this doesn't have `eth_getProof`!

Comment (jeffsmale90) [reply .github/pull_request_template.md:22]: https://github.com/trufflesuite/ganache/issues/3972

**Review by tenthirtyone — APPROVED** [review]
:+1: 

I usually see two different types of PR template. This kind and a 'preflgiht pr checklist' is how I think of the 2nd.

https://github.com/devspace/awesome-github-templates

I like these a lot for starting a funnel before we triage issues, too, if it works out for PRs.

**Comment by MicaiahReid** [comment]
I've rearranged a bit and added some further comments to make very clear that most of the information is optional for external contributors. @davidmurdoch's concern is that these templates will be a barrier to entry or an annoyance for some contributors, and we definitely don't want that. Input welcome.

**Review by tenthirtyone — APPROVED** [review]

**Comment by davidmurdoch** [comment]
I don't like it :-/

**Comment by MicaiahReid** [comment]
Seems like this isn't something we think is best for us right now.

**Pull request closed without merging**